packages/block-library/src/navigation-link/test/edit.js

```diff
@@ -0,0 +1,506 @@
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				title: 'Cats',
+				type: 'category',
+				url: 'http://wordpress.local/category/cats/',
+			};
+			updateNavigationLinkBlockAttributes(
+				linkSuggestion,
+				setAttributes
+			);
+			expect( setAttributes ).toHaveBeenCalledWith( {
+				id: 9,
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				label: 'Cats',
+				type: 'category',
+				url: 'http://wordpress.local/category/cats/',
+			} );
+		} );
+
+		it( 'can update a custom post type link', () => {
+			const setAttributes = jest.fn();
+			const linkSuggestion = {
+				id: 131,
+				kind: 'post-type',
+				opensInNewTab: false,
+				title: 'Fall',
+				type: 'portfolio',
+				url: 'http://wordpress.local/portfolio/fall/',
+			};
+			updateNavigationLinkBlockAttributes(
+				linkSuggestion,
+				setAttributes
+			);
+			expect( setAttributes ).toHaveBeenCalledWith( {
+				id: 131,
+				kind: 'post-type',
+				opensInNewTab: false,
+				label: 'Fall',
+				type: 'portfolio',
+				url: 'http://wordpress.local/portfolio/fall/',
+			} );
+		} );
+
+		it( 'can update a custom tag link', () => {
+			const setAttributes = jest.fn();
+			const linkSuggestion = {
+				id: 4,
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				title: 'PortfolioTag',
+				type: 'portfolio_tag',
+				url: 'http://wordpress.local/portfolio_tag/portfoliotag/',
+			};
+			updateNavigationLinkBlockAttributes(
+				linkSuggestion,
+				setAttributes
+			);
+			expect( setAttributes ).toHaveBeenCalledWith( {
+				id: 4,
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				label: 'PortfolioTag',
+				type: 'portfolio_tag',
+				url: 'http://wordpress.local/portfolio_tag/portfoliotag/',
+			} );
+		} );
+
+		it( 'can update a custom category link', () => {
+			const setAttributes = jest.fn();
+			const linkSuggestion = {
+				id: 2,
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				title: 'Portfolioz Category',
+				type: 'portfolio_category',
+				url:
+					'http://wordpress.local/portfolio_category/portfolioz-category/',
+			};
+			updateNavigationLinkBlockAttributes(
+				linkSuggestion,
+				setAttributes
+			);
+			expect( setAttributes ).toHaveBeenCalledWith( {
+				id: 2,
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				label: 'Portfolioz Category',
+				type: 'portfolio_category',
+				url:
+					'http://wordpress.local/portfolio_category/portfolioz-category/',
+			} );
+		} );
+
+		it( 'can update a post format', () => {
+			const setAttributes = jest.fn();
+			const linkSuggestion = {
+				id: 'video',
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				title: 'Video',
+				type: 'post-format',
+				url: 'http://wordpress.local/type/video/',
+			};
+			updateNavigationLinkBlockAttributes(
+				linkSuggestion,
+				setAttributes
+			);
+			expect( setAttributes ).toHaveBeenCalledWith( {
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				label: 'Video',
+				type: 'post_format',
+				url: 'http://wordpress.local/type/video/',
+			} );
```

Comment:
I wonder whether we should have an explicit assertion around `id` _not_ being set when the linkSuggestion `id` is not an int (as in this case).